pkg/storage/gcp/bucket_mock.go

```diff
@@ -0,0 +1,95 @@
+package gcp
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+)
+
+type bucketMock struct {
+	db   map[string][]byte
+	lock sync.RWMutex
+}
+
+func newBucketMock() Bucket {
+	b := &bucketMock{}
+	b.db = make(map[string][]byte)
+	return b
+}
+
+type bucketReader struct {
+	io.Reader
+	*bucketMock
+}
+
+type bucketWriter struct {
+	*bucketMock
+	path string
+}
+
+func (r *bucketReader) Close() error {
+	r.bucketMock.lock.RUnlock()
+	return nil
+}
+
+func (r *bucketWriter) Close() error {
+	r.bucketMock.lock.Unlock()
+	return nil
+}
+
+func (r *bucketWriter) Write(p []byte) (int, error) {
+	_, ok := r.bucketMock.db[r.path]
+	if !ok {
+		r.bucketMock.db[r.path] = make([]byte, 0)
+	}
+
+	r.bucketMock.db[r.path] = append(r.bucketMock.db[r.path], p...)
+
+	return len(p), nil
+}
+
+func (m *bucketMock) Delete(ctx context.Context, path string) error {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	delete(m.db, path)
+	return nil
+}
+
+func (m *bucketMock) Open(ctx context.Context, path string) (io.ReadCloser, error) {
```

Comment:
does this need to use a read lock before you do the lookup?

Reply:
You are right, it does. Will change it